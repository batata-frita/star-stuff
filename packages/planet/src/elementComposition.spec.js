const elementComposition = require('./elementComposition')

const weightedElements = [
  {
    code: 'H',
    weight: 60
  },
  {
    code: 'He',
    weight: 20
  },
  {
    code: 'O',
    weight: 5
  },
  {
    code: 'C',
    weight: 15
  }
]

test('elementComposition', () => {
  const composition = elementComposition(weightedElements)(0.5)

  expect(composition).toMatchSnapshot()
})
