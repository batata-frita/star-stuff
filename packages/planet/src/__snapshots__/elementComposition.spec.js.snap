// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elementComposition 1`] = `
Array [
  Object {
    "code": "H",
    "ratio": 0.4523767806596055,
  },
  Object {
    "code": "He",
    "ratio": 0.4655329424761272,
  },
  Object {
    "code": "O",
    "ratio": 0.0771970743291619,
  },
  Object {
    "code": "C",
    "ratio": 0.004893202535105416,
  },
]
`;
